core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -318,6 +319,26 @@ public void testFromXContent() throws IOException {
     }
 
     /**
+     * Test that unknown field trigger ParsingException.
+     * To find the right position in the root query, we add a marker as `queryName` which
+     * all query builders support. The added bogus field after that should trigger the exception.
+     * Queries that allow arbitrary field names at this level need to override this test.
+     */
+    public void testUnknownField() throws IOException {
+        QB testQuery = createTestQueryBuilder();
+        assumeFalse("Skip test in unlikely event that we have _marker_ in some random string", testQuery.toString().contains("_marker_"));
```

Comment:
Ah! Well if its a huge pain then its probably not worth it. Its just that if you do something drastic one of the simplest metrics to make sure you didn't break anything is to count the number of tests. And if some tests skip sometimes and not other times you have to manually validate those tests. Its not worth optimizing for this case though if its difficult.